packages/x-components/src/x-modules/queries-preview/components/__tests__/query-preview.spec.ts

```diff
@@ -101,8 +103,77 @@ describe('query preview', () => {
     expect(getXComponentXModuleName(wrapper.vm)).toBe('queriesPreview');
   });
 
+  // eslint-disable-next-line max-len
+  it('does not send the `QueryPreviewRequestUpdated` event if persistInCache is true, but emits load', () => {
+    const { queryPreviewRequestUpdatedSpy, wrapper } = renderQueryPreview({
+      persistInCache: true,
+      queryPreviewInfo: {
+        query: 'shoes',
+        extraParams: { directory: 'Magrathea' },
+        filters: ['fit:regular']
+      }
+    });
+
+    jest.advanceTimersByTime(0); // Wait for first emission.
+    expect(queryPreviewRequestUpdatedSpy).toHaveBeenCalledTimes(0);
+    expect(wrapper.emitted('load')?.length).toBe(1);
+    expect(wrapper.emitted('load')?.[0]).toEqual(['shoes']);
+  });
+
+  it('does not remove the query before destroying the component', () => {
+    const { wrapper } = renderQueryPreview({
+      persistInCache: true,
+      queryPreviewInfo: {
+        query: 'shoes',
+        extraParams: { directory: 'Magrathea' },
+        filters: ['fit:regular']
+      }
+    });
+
+    wrapper.destroy();
+    expect(
+      wrapper.vm.$store.getters['x/queriesPreview/loadedQueriesPreview']['shoes']
+    ).toBeTruthy();
+    expect(wrapper.emitted('load')?.length).toBe(1);
+    expect(wrapper.emitted('load')?.[0]).toEqual(['shoes']);
+  });
+
+  it('removes the query before destroying the component', () => {
+    const { queryPreviewRequestUpdatedSpy, wrapper } = renderQueryPreview({
+      persistInCache: false,
+      queryPreviewInfo: {
+        query: 'shoes',
+        extraParams: { directory: 'Magrathea' },
+        filters: ['fit:regular']
+      }
+    });
+
+    jest.advanceTimersByTime(0); // Wait for first emission
+    wrapper.destroy();
+    expect(wrapper.vm.$store.getters['x/queriesPreview/loadedQueriesPreview']['shoes']).toBeFalsy();
+    expect(queryPreviewRequestUpdatedSpy).toHaveBeenCalledTimes(1);
+    expect(queryPreviewRequestUpdatedSpy).toHaveBeenCalledWith({
+      extraParams: {
+        directory: 'Magrathea'
+      },
+      filters: {
+        fit: [
+          {
+            id: 'fit:regular',
+            modelName: 'RawFilter',
+            selected: true
+          }
+        ]
+      },
+      origin: undefined,
+      query: 'shoes',
+      rows: 24
+    });
+  });
+
```

Comment:
We're testing a whole flow instead of just the component. We shouldn't care about what the app does when we emit the event to remove the query preview, we just want to tests that it does emit the event.
This way, if the store implementation fails or changes for whatever reason, these tests aren't affected.